regex-syntax/Cargo.toml

```diff
@@ -9,5 +9,12 @@ homepage = "https://github.com/rust-lang/regex"
 description = "A regular expression parser."
 workspace = ".."
 
+[features]
+default = ["std"]
+# Disable this on-by-default feature and add "alloc" to allow use in no_std builds
+std = []
+# Required for use in no_std builds, presently nightly-only
+alloc = []
```

Comment:
I don't think you need an `alloc` feature flag if it will always be required — i.e. unless there are plans for core-only support in the future you could just drop this feature flag and do `#![cfg_attr(not(feature = "std"), feature(alloc))]`.

Reply:
I'd like to leave the door open for that possibility. We can simplify the cfg statements if those plans don't materialize.

Reply:
Trick we do with Rand: make `std` depend on `alloc`. This has the advantage that you can use just `#[cfg(feature = "alloc")]` to feature-gate modules/items requiring an allocator (though also the disadvantage that sometimes you need to check both: `#![cfg_attr(all(feature="alloc", not(feature="std")), feature(alloc))]`).